grammar.js

```diff
@@ -876,18 +882,18 @@ module.exports = grammar({
 
     continue_statement: $ => seq('continue', ';'),
 
-    do_statement: $ => seq('do', $._statement, 'while', '(', $._expression, ')', ';'),
+    do_statement: $ => seq('do', field('statement', $._statement), 'while', '(', field('condition', $._expression), ')', ';'),
 
     empty_statement: $ => ';',
 
     expression_statement: $ => seq($._expression_statement_expression, ';'),
 
-    fixed_statement: $ => seq('fixed', '(', $.variable_declaration, ')', $._statement),
+    fixed_statement: $ => seq('fixed', '(', field('declaration', $.variable_declaration), ')', $._statement),
 
     for_statement: $ => seq(
       'for',
       '(',
-      field('initializer', optional(choice($.variable_declaration, commaSep1($._expression)))),
+      field('initializer', optional(choice(field('declaration', $.variable_declaration), commaSep1($._expression)))),
```

Comment:
I am not sure what the effect of nesting is here. Perhaps something like this is less ambiguous:

```suggestion
      optional(choice(field('declaration', $.variable_declaration), commaSep1(field('initializer', $._expression))))),
```